crates/crypto/proof-params/src/lib.rs

```diff
@@ -14,15 +15,55 @@ pub use traits::{
     DummyWitness, ProvingKeyExt, VerifyingKeyExt,
 };
 
-#[cfg(feature = "proving-keys")]
+#[cfg(feature = "bundled-proving-keys")]
 mod proving_keys;
 
+#[derive(Debug, Default)]
+pub struct LazyProvingKey {
+    inner: OnceCell<ProvingKey<Bls12_377>>,
+}
+
+impl LazyProvingKey {
+    pub fn new() -> Self {
+        LazyProvingKey {
+            inner: OnceCell::new(),
+        }
+    }
+
+    pub fn load(&self, bytes: &[u8]) -> ProvingKey<Bls12_377> {
+        self.inner
+            .get_or_init(|| {
+                ProvingKey::deserialize_uncompressed_unchecked(bytes)
+                    .expect("Failed to deserialize ProvingKey!")
+            })
+            .to_owned()
```

Comment:
How does this `load` method relate to the existing `load_proving_parameters`? That method checks the provided data against hardcoded hashes of the proving key. Could we combine them?

Reply:
I think we could do it if we made the `LazyProvingKey` take a `&'static str`!

Reply:
Fixed in cb83595